src/main/java/com/iotiq/interview/service/CategoryService.java

```diff
@@ -35,4 +35,11 @@ public Category create(CategoryRequest request) {
     public List<Category> getAll() {
         return categoryRepository.findAll();
     }
-}
+
+    public List<Category> getFiltered(String name) {
```

Comment:
Say we have 15 fields for categories and we want to use 13 of them for filtering. How do you do that? Please change your implementation that can be extended for such a case.